### PR TITLE
i18n: update doc about vue-i18n@6.x

### PR DESCRIPTION
**Pull request by QingWei-Li**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: examples/docs/en-US/i18n.md [hunk 1]
```diff
@@ -72,6 +72,42 @@ Vue.use(Element, {
 })
 ```
 
+## Compatibility `vue-i18n@6.x`
```
Comment (Leopoldthecoder) [inline examples/docs/en-US/i18n.md]: 和上面保持一致吧，Compatible with

**Comment by BelinChung** [comment]
How do I configure eui components language when I importing Element on demand and vue-i18n@6.x. It doesn't seem to mention in the document?
当我使用 `vue-i18n@6.x`， 我该如何配置按需引用的 eui 组件的国际化语言，它好像没有在文档中提及？

**Comment by QingWei-Li** [comment]
@BelinChung 
```js
import Vue from 'vue'
import Element from 'element-ui'
import VueI18n from 'vue-i18n'
import enLocale from 'element-ui/lib/locale/lang/en'
import zhLocale from 'element-ui/lib/locale/lang/zh-CN'
import App from './App.vue'

import localeProvider from 'element-ui/lib/locale'

Vue.use(VueI18n)

const messages = {
  en: {
    message: 'hello',
    ...enLocale // 或者用 Object.assign({ message: 'hello' }, enLocale)
  },
  zh: {
    message: '你好',
    ...zhLocale // 或者用 Object.assign({ message: '你好' }, zhLocale)
  }
}
// Create VueI18n instance with options
const i18n = new VueI18n({
  locale: 'en', // set locale
  messages, // set locale messages
})

localeProvider.i18n(key => i18n.t(key))
Vue.use(Element.DatePicker)

new Vue({ i18n, render: h => h(App) }).$mount('#app')
```

**Comment by BelinChung** [comment]
@QingWei-Li 谢谢！

**Comment by LoverMaker** [comment]
```
import Vue from 'vue';
import ElementUI from 'element-ui';
// ele语言包
import enLocale from 'element-ui/lib/locale/lang/en';
import zhLocale from 'element-ui/lib/locale/lang/zh-CN';
// 自定义语言包
import en from './local/lang/en';
import zhCN from './local/lang/zh-CN';

import format from 'element-ui/lib/locale/format';
const formater = format(Vue);
// 设置语言
Vue.use(VueI18n);
// 语言包
const messages = {
  en: {
    ...en,
    ...enLocale
  },
  zhCN: {
    ...zhCN,
    ...zhLocale
  }
};
// vue-i18n
const i18n = new VueI18n({
  locale: 'zhCN', // set locale
  messages, // set locale messages
});
// 初始化ElementUI语言包
Vue.use(ElementUI, {
  i18n(path, options) {
    let value = i18n.t(path);
    if (value !== null && value !== undefined) return formater(value, options);
    const array = path.split('.');
    for (let i = 0, j = array.length; i < j; i += 1) {
      const property = array[i];
      value = i18n.t(property);
      if (i === j - 1) return formater(value, options);
      if (!value) return '';
    }
    return '';
  }
});
new Vue({ i18n, render: h => h(App) }).$mount('#app')
```
这里必须需要进行format处理吧，有的key是`el.total.pagination: 'Total {total}'`， {total}需要替换！

**Pull request merged**